server_modules/address.js

```diff
@@ -0,0 +1,19 @@
+var os = require('os');
+var args = require('./args')
+
+//shamelessly copied from StackOverflow with only minor changes
+var interfaces = os.networkInterfaces();
+var addresses = [];
+for (var i in interfaces) {
+    for (var j in interfaces[i]) {
+        var address = interfaces[i][j];//we basically parse over all the addresses for every network interface
+        if (address.family === 'IPv4' && !address.internal) {//if that address is an IPv4 address, and it is not an internal one, we add it
+            addresses.push(address.address);
+        }
```

Comment:
You only use the first you find, but you continue to iterate over all of them. Use break or add the condition into the loop condition. Also you can keep only one in memory, but that's not really important